src/mapbox/mapbox.js

```diff
@@ -39,6 +39,7 @@ const propTypes = {
   attributionControl: PropTypes.bool, /** Show attribution control or not. */
   preserveDrawingBuffer: PropTypes.bool, /** Useful when you want to export the canvas as a PNG. */
   onLoad: PropTypes.func, /** The onLoad callback for the map */
+  onError: PropTypes.func, /** The onError callback for the map */
```

Comment:
Outside the scope of this, but do we want to allow external overrides / ability to hook into these?

Reply:
Mapbox's full list of events (removed mouse/gesture events):
- resize
- remove
- webglcontextlost
- webglcontextrestored
- load
- render
- error
- data
- styledata
- sourcedata
- dataloading
- styledataloading
- sourcedataloading

Now that we expose `getMap` as a public method, the user is able to register their own listeners if they need to. Aside from `load` and `error`, I've also used `render` and `dataloading`. Not sure if all of them are useful in the React context.

Reply:
That makes sense. Would registering them on the Mapbox instance override our event listener for those events, though?

Not a blocking thing for this PR btw, feel free to land whenever you want.